### PR TITLE
Scala 2.13

### PR DESCRIPTION
**Pull request by e-orz**
Let's do it!

### CODE REVIEW

File: server/cmwell-util/src/main/scala/cmwell/util/files/package.scala [hunk 1]
```diff
@@ -52,12 +52,14 @@ package object files extends LazyLogging {
 
   }
 
+/*
```
Comment (MoriaYechiel) [inline server/cmwell-util/src/main/scala/cmwell/util/files/package.scala]: Throw away

File: server/cmwell-ws/app/actions/ActiveInfotonGenerator.scala [hunk 2]
```diff
@@ -494,22 +494,13 @@ class ActiveInfotonGenerator @Inject()(
...
-        (String, Color.Color),
-        (String, Color.Color)))
-    ], (String, Set[FieldValue]), Map[String, Set[FieldValue]]]
   private[this] def generateHealthDetailedFields: FieldsOpt = {
     //val (xs, timeStamp) = DashBoardCache.cacheAndGetDetailedHealthData
     val res = getClusterDetailedHealthNew
     Some(
       res.toSeq
         .sortBy(_._1)
+        .view
```
Comment (MoriaYechiel) [inline server/cmwell-ws/app/actions/ActiveInfotonGenerator.scala]: Consider to refactor

File: server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala [hunk 3]
```diff
@@ -370,7 +360,7 @@ class ImpStream(partition: Int,
                           infoton.path.getBytes,
                           payload
               )
-            }(breakOut3))
+            }.to(IIterable))
```
Comment (MoriaYechiel) [inline server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala]: Isn't there too many I's ?

File: server/cmwell-cons/src/main/scala/Deployment.scala [hunk 4]
```diff
@@ -475,9 +476,9 @@ case class BgProps(h: Host)
   override val componentName: String = "cmwell-bg"
   override val componentMappings: Map[String, Int] = Map("bg" -> 1)
 
-  override def stop(hosts: GenSeq[String]): Unit = h.stopBg(hosts)
+  override def stop(hosts: ParSeq[String]): Unit = h.stopBg(hosts)
 
-  override def start(hosts: GenSeq[String]): Unit = h.startBg(hosts)
+  override def start(hosts: Seq[String]): Unit = h.startBg(hosts.to(ParSeq))
```
Comment (MoriaYechiel) [inline server/cmwell-cons/src/main/scala/Deployment.scala]: Why they're both GenSeq and you changed one to ParSeq and the other to Seq?

File: server/cmwell-cons/src/main/scala/ctl.scala [hunk 5]
```diff
@@ -349,8 +349,18 @@ abstract class Host(user: String,
...
-    if (sudo)
-      command(s"""echo -e '$content' | sudo tee $path > /dev/null""", host, true, sudoer)
+    if (sudo) {
+      val tempFile = File.createTempFile("pre-", ".txt")
+      val fullPath = tempFile.getAbsolutePath
+      val fileName = tempFile.getName
+      val writer = new PrintWriter(tempFile)
+      writer.write(content)
+      writer.close()
+      _rsync(fullPath, "~", s"${sudoer.get.name}@$host", sudo = false)
```
Comment (MoriaYechiel) [inline server/cmwell-cons/src/main/scala/ctl.scala]: Change to:
command(s"""echo $$'$content' > $path""", host, false)

File: server/cmwell-cons/src/main/scala/ctl.scala [hunk 6]
```diff
@@ -1416,32 +1425,30 @@ abstract class Host(user: String,
   def startScript(script: String) =
     s"""HAL=9000 ${if (deb) "CMWELL_DEBUG=true" else ""} $script"""
 
-  def start: Unit = start(ips.par)
-
-  def start(hosts: String*): Unit = start(hosts.par)
+  def start: Unit = start(ips)
 
-  def start(hosts: GenSeq[String]) {
+  def start(hosts: Seq[String]) {
```
Comment (MoriaYechiel) [inline server/cmwell-cons/src/main/scala/ctl.scala]: hostsPArSeq = hosts.to(ParSeq)

File: server/cmwell-cons/src/main/scala/ctl.scala [hunk 7]
```diff
@@ -1504,19 +1511,19 @@ abstract class Host(user: String,
     startKafka
 
     info("  inserting schemas")
-    initSchemes(hosts)
+    initSchemes(hosts.to(ParSeq))
```
Comment (MoriaYechiel) [inline server/cmwell-cons/src/main/scala/ctl.scala]: hostsPArSeq = hosts.to(ParSeq)

File: server/cmwell-cons/src/main/scala/ctl.scala [hunk 8]
```diff
@@ -1675,40 +1682,36 @@ abstract class Host(user: String,
...
     checkProduction
   }
 
-  def install: Unit = install(ips.par)
-
-  def install(hosts: String*): Unit = install(hosts.par)
+  def install: Unit = install(ips)
 
-  def install(hosts: GenSeq[String]) {
+  def install(hosts: Seq[String]) {
```
Comment (MoriaYechiel) [inline server/cmwell-cons/src/main/scala/ctl.scala]: hostsPArSeq = hosts.to(ParSeq)

File: server/project/cmwell-build-plugin/src/main/scala/cmwell/build/CMWellBuild.scala [hunk 9]
```diff
@@ -236,7 +236,7 @@ object CMWellBuild extends AutoPlugin {
 	override def requires = PackPlugin && ScalastylePlugin /*&& ScalafmtPlugin*/ && DoctestPlugin /*&& DependencyGraphPlugin*/
 
 	override def projectSettings = Seq(
-		scalastyleFailOnError := true,
+		scalastyleFailOnError := false,
```
Comment (MoriaYechiel) [inline server/project/cmwell-build-plugin/src/main/scala/cmwell/build/CMWellBuild.scala]: Why??????

File: server/cmwell-util/src/main/scala/cmwell/util/files/package.scala (hunk 1 above)
Comment (e-orz) [reply server/cmwell-util/src/main/scala/cmwell/util/files/package.scala]: done

File: server/cmwell-ws/app/actions/ActiveInfotonGenerator.scala (hunk 2 above)
Comment (e-orz) [reply server/cmwell-ws/app/actions/ActiveInfotonGenerator.scala]: Sort must have all the elements. I couldn't find a way to sort directly on a Map.

File: server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala (hunk 3 above)
Comment (e-orz) [reply server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala]: No, it's an alias for immutable.Iterable. See the imports at the beginning of the file.

File: server/cmwell-cons/src/main/scala/Deployment.scala (hunk 4 above)
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/Deployment.scala]: The hosts is a normal Seq (ES need the ordering). StartBg method takes ParSeq.

File: server/cmwell-cons/src/main/scala/ctl.scala (hunk 5 above)
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/ctl.scala]: Tried it. It worked for the `60-cassandra.conf` but no for the `cmwell-runner`. Leaving it as it is now.

File: server/cmwell-cons/src/main/scala/ctl.scala (hunk 6 above)
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/ctl.scala]: done

File: server/cmwell-cons/src/main/scala/ctl.scala (hunk 7 above)
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/ctl.scala]: done

File: server/cmwell-cons/src/main/scala/ctl.scala (hunk 8 above)
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/ctl.scala]: done

File: server/cmwell-cons/src/main/scala/ctl.scala [hunk 10]
```diff
@@ -1922,10 +1925,10 @@ abstract class Host(user: String,
     currentVersion.map(ver => info(s"Current version is $ver"))
 
     checkProduction
-    refreshUserState(user, None, hosts)
+    refreshUserState(user, None, hosts.to(ParSeq))
```
[the remark this answers is not in the document]
Comment (e-orz) [reply server/cmwell-cons/src/main/scala/ctl.scala]: done

File: server/cmwell-ws/app/BGMonitorActor.scala [hunk 11]
```diff
@@ -41,18 +43,15 @@ class BGMonitorActor(zkServers: String,
...
-    }
-    .mkString(",")
+  val kafkaAdminProperties = new Properties
+  kafkaAdminProperties.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaURL)
+  val adminClient = AdminClient.create(kafkaAdminProperties)
   val topics = Seq("persist_topic", "persist_topic.priority", "index_topic", "index_topic.priority")
-  val partitionsForTopics = zkUtils.getPartitionsForTopics(topics)
-  val topicsPartitionsAndGroups = partitionsForTopics.flatMap {
+  val describedTopics = adminClient.describeTopics(topics.asJava).all().get(30, TimeUnit.SECONDS)
+  val partitionsForTopics2: mutable.Map[String, Seq[Int]] = describedTopics.asScala.map {
```
[the remark this answers is not in the document]
Comment (e-orz) [reply server/cmwell-ws/app/BGMonitorActor.scala]: done

File: server/project/cmwell-build-plugin/src/main/scala/cmwell/build/CMWellBuild.scala (hunk 9 above)
Comment (e-orz) [reply server/project/cmwell-build-plugin/src/main/scala/cmwell/build/CMWellBuild.scala]: scalastyle still doesn't support Scala 2.13

**Pull request merged**